lib/dialogs.dart

```diff
@@ -52,6 +52,82 @@ class AboutDialog extends DDialog {
   }
 }
 
+class ExportDialog extends DDialog {
+  final GistContainer gistContainer;
+  final GistController gistController;
+
+  ParagraphElement _text;
+  ParagraphElement _controls;
+  TextAreaElement _textArea;
+  DButton _cancelButton;
+  DButton _htmlButton;
+  DButton _docButton;
+  DButton _inlineButton;
+  
+  ExportDialog(
+      GistContainer this.gistContainer, GistController this.gistController)
+      : super(title: 'Export') {
+    element.classes.toggle('sharing-dialog', true);
+
+    content.setAttr('layout');
+    content.setAttr('vertical');
+
+    _text = content.add(new ParagraphElement());
+    _controls = content.add(new ParagraphElement());
+    _textArea = content.add(new TextAreaElement());
+    _textArea.className = 'sharingSummaryText';
+    _textArea.setAttribute('flex', '');
+
+    _cancelButton = new DButton.button(text: 'Cancel');
```

Comment:
Let's remove these three buttons (just have a close button), and automatically generate all three links into the text area when the dialog opens:

```
dart: https://dartpad.dartlang.org/embed-dart.html?id=76567ABC
dart + html: https://dartpad.dartlang.org/embed-html.html?id=76567ABC
dart (minimal): https://dartpad.dartlang.org/embed-inline.html?id=76567ABC
```